zebra-chain/src/block/height.rs

```diff
@@ -41,6 +43,54 @@ impl Height {
     pub const MAX_AS_U32: u32 = Self::MAX.0;
 }
 
+// Block heights live in a torsor, they can only be contructed but not computed directly.
+// Addition and substraction is done in the underlying types(u32) and not in the heights themselves.
+// This u32 numbers are a group and they can be added and substracted normally to get a new integer.
+// Having the result of the computation the `Height(X)` constructor can be used to build a new `Height`.
+
+// N = Height1.U - Height2.U
+// Given two `Height`s use the underlying value of them to compute the difference,
+// return this number as it is.
+impl Sub<Height> for Height {
+    type Output = Option<u32>;
+
+    fn sub(self, rhs: Height) -> Option<u32> {
+        self.0.checked_sub(rhs.0)
+    }
+}
```

Comment:
Rather than using an `u32`, this and the other impls should use an `i32`, because one block height might be bigger than the other, giving a negative value.  There's a list of impl specifications in #1055. 

Because the inner `u32` of a `Height` is in the range `0 .. 500_000_000`, we know there's no risk of underflow, so the `Output` type can be an `i32` rather than an `Option`, and the subtraction can just be

```
(self.0 as i32) - (rhs.0 as i32)
```

Reply:
Changed this at https://github.com/ZcashFoundation/zebra/pull/1057/commits/4e3a77479985a2bce82a4cd9fb0d138ecda2fca4